### ISSUE TITLE
Block version of `source` in Gemfile does not work with RUBYGEMS_GEMDEPS

### ISSUE THREAD

**Issue by mattbrictson** [rubygems/rubygems#1210]
It looks like Rubygems doesn't support the Gemfile DSL where `source` is given a block. When I use `RUBYGEMS_GEMDEPS=-` with a Gemfile that has a source block, I get `undefined method 'version' for nil:NilClass (NoMethodError)` (full stack trace below).

For example, here is a trivial Gemfile that demonstrates the problem:

```ruby
source "https://rails-assets.org" do
  gem "rails-assets-moment"
end
```

And the Gemfile.lock produced by bundler 1.9.1:

```
GEM
  remote: https://rails-assets.org/
  specs:
    rails-assets-moment (2.9.0)

PLATFORMS
  ruby

DEPENDENCIES
  rails-assets-moment!
```

Trying to activate these dependencies (i.e. using `RUBYGEMS_GEMDEPS=-`) results in the following error:

```
rubygems/request_set/lockfile/parser.rb:324:in `pinned_requirement': undefined method `version' for nil:NilClass (NoMethodError)
  from rubygems/request_set/lockfile/parser.rb:80:in `parse_DEPENDENCIES'
  from rubygems/request_set/lockfile/parser.rb:22:in `parse'
  from rubygems/request_set.rb:282:in `load_gemdeps'
  from rubygems.rb:1055:in `use_gemdeps'
  from rubygems.rb:1244:in `<top (required)>'
  from <internal:gem_prelude>:1:in `require'
  from <internal:gem_prelude>:1:in `<compiled>'
```

Ruby 2.2.1p85, Rubygems 2.4.6.

**Comment by aripollak** [comment]
This appears to be fixed in the latest version of rubygems.

**Comment by mattbrictson** [comment]
Yes, seems to be fixed in 2.5.2. Thanks for checking!
